frontend/lib/resolution/prims.cpp

```diff
@@ -374,6 +374,36 @@ static QualifiedType primTypeof(Context* context, PrimitiveTag prim, const CallI
   return QualifiedType(QualifiedType::TYPE, typePtr);
 }
 
+static QualifiedType primGetSvecMember(Context* context, PrimitiveTag prim,
+                                       const CallInfo& ci) {
+  CHPL_ASSERT(prim == PRIM_GET_SVEC_MEMBER ||
+              prim == PRIM_GET_SVEC_MEMBER_VALUE);
+
+  if (ci.numActuals() == 2 && ci.actual(0).type().hasTypePtr()) {
+    auto index = ci.actual(1).type();
+    if (index.hasTypePtr() && index.type()->isIntegralType()) {
+      auto act = ci.actual(0).type();
+      if (auto tup = act.type()->toTupleType()) {
+        if (tup->isStarTuple()) {
+          auto eltType = tup->starType();
+          QualifiedType::Kind retKind = eltType.kind();
+          if (prim == PRIM_GET_SVEC_MEMBER_VALUE) {
+            // Must return as value for _VALUE variant, but still maintain
+            // constness and param-ness as applicable.
+            if (!eltType.isParam()) {
+              retKind = (eltType.isConst() ? QualifiedType::CONST_VAR
+                                           : QualifiedType::VAR);
+            }
+          }
+          return QualifiedType(retKind, eltType.type());
```

Comment:
This seems like a use case for the `KindProperties` struct in `can-pass.cpp`. That struct is intend to decompose an intent into its "varness, refness, etc", manipulate it, and put it back together. You could probably get the behavior you desire here by combining `retKind` with the `PARAM` intent (the result will be param if the other kind was `PARAM`, `CONST` if `retKind` is `CONST`, and `VAR` otherwise). 

Consider adding a static method to `KindProperties`:

```C++
static KindProperties combineKinds(types::QualifiedType::Kind kind1, types::QualifiedType::Kind kind1) {
  auto kp1 = KIndProperties::fromKind(kind1);
  auto kp2 = KIndProperties::fromKind(kind2);
  return kp1.combineWith(kp2).toKind();
}
```

Then this could be:

```suggestion
          QualifiedType::Kind retKind = eltType.kind();
          if (prim == PRIM_GET_SVEC_MEMBER_VALUE) {
            retKind = KindProperties::combineKinds(retKind, QualifiedType::PARAM);
          }
          return QualifiedType(retKind, eltType.type());
```

Reply:
Hmm, perhaps some sort of `combineWithStrict` would need to work instead to handle `param`s.